data/storeDEPRECATED/mongo/mongo_bucket_store_client.go

```diff
@@ -0,0 +1,148 @@
+	client.StoreClient = store
+	return &client, err
+}
+
+/* bucket methods */
+
+// Look for a single bucket based on its Id
+func (c *MongoBucketStoreClient) Find(ctx context.Context, bucket *schema.CbgBucket) (result *schema.CbgBucket, err error) {
+	
+	if bucket.Id != "" {
+		var query bson.M = bson.M{}
+		tid, _ := primitive.ObjectIDFromHex(bucket.Id)
+		query["_id"] = tid
+		opts := options.FindOne()
+		opts.SetSort(bson.D{primitive.E{Key: "_id", Value: -1}})
+		if err = c.Collection("hotDailyCbg").FindOne(ctx, query, opts).Decode(&result); err != nil && err != mongo.ErrNoDocuments {
+			c.log.WithError(err)
+			return result, err
+		}
+	
+		return result, nil
+	}
+
+	return nil, errors.New("Find called with an empty bucket.Id")
+}
+
+// Update a bucket record if found overwhise it will be created. The bucket is searched by its id.
+func (c *MongoBucketStoreClient)Upsert(ctx context.Context, userId *string, creationTimestamp time.Time, sample *schema.CbgSample) error {
+	
+	if sample == nil {
+		return errors.New("impossible to upsert a nil sample")
+	}
+
+	if sample.TimeStamp.IsZero() {
+		return errors.New("impossible to upsert a sample having a incorrect timestamp")
+	}
+
+	if userId == nil {
+		return errors.New("impossible to upsert a sample for an empty user id")
+	}
+
+	// Extrat ISODate from sample timestamp
+	ts := sample.TimeStamp.Format("02-01-2006")
+	valTrue := true
+	strUserId := *userId
+
+	c.log.Info("upsert cbg sample for: " + strUserId + "_" + ts)
+	// save in hotDailyCbg
+	_, err := c.Collection("hotDailyCbg").UpdateOne(
+		ctx,
+		bson.D{{Key: "_id", Value: strUserId + "_" + ts }}, // filter
+		bson.D{ // update
+			{Key: "$addToSet", Value: bson.D{{Key: "samples", Value: sample }}},
+			{Key: "$setOnInsert", Value: bson.D{{Key: "_id", Value: strUserId + "_" + ts }}},
+			{Key: "$setOnInsert", Value: bson.D{{Key: "creationTimestamp", Value: creationTimestamp }}},
+			{Key: "$setOnInsert", Value: bson.D{{Key: "day", Value: ts }}},
+		},
+		&options.UpdateOptions{Upsert: &valTrue}, //options
+	)
+
+	if err != nil {
+		return err
+	}
+
+	// save in coldDailyCbg
+	_, err = c.Collection("coldDailyCbg").UpdateOne(
+		ctx,
+		bson.D{{Key: "_id", Value: strUserId + "_" + ts }}, // filter
+		bson.D{ // update
+			{Key: "$addToSet", Value: bson.D{{Key: "samples", Value: sample }}},
+			{Key: "$setOnInsert", Value: bson.D{{Key: "_id", Value: strUserId + "_" + ts }}},
+			{Key: "$setOnInsert", Value: bson.D{{Key: "creationTimestamp", Value: creationTimestamp }}},
+			{Key: "$setOnInsert", Value: bson.D{{Key: "day", Value: ts }}},
+		},
+		&options.UpdateOptions{Upsert: &valTrue}, //options
+	)
+
+	return err
+}
+
+// Perform a bulk of operations on bucket records based on the operation argument, update a record if found overwhise created ot. 
+// The bucket is searched by its id.
+func (c *MongoBucketStoreClient)UpsertMany(ctx context.Context, userId *string, operations []mongo.WriteModel) error {
```

Comment:
why not doing the write model inside the method ? 
So that you pass here only an array of samples (easy to use) and the bulk complexity is hidden in this method (otherwise each caller need to create WriteModel)